include/endianshim.h

```diff
@@ -13,6 +13,15 @@
 #define le16toh(x) OSSwapLittleToHostInt16(x)
 #endif
 
+#ifdef __SUNOS
+#include <sys/byteorder.h>
+#ifdef _BIG_ENDIAN
+#define le16toh(x) BSWAP_16(x)
+#else
+#define le16toh(x) (x)
+#endif
+#endif
```

Comment:
Unfortunately Solaris needs a different header.  I updated the PR with your recent changes and used the builtin BSWAP_16 macro on this platform.

Thanks!